### PR TITLE
Add `scipy.special.spherical_yn`

### PR DESCRIPTION
**Pull request by chettub**
This PR adds `scipy.special.spherical_yn`.

This implementation supports real numbers and uses almost the same algorithm as in scipy.
Calculation for complex number requires other unimplemented Bessel functions, and therefore not supported in this PR.

Linked Issue : #6324

### CODE REVIEW

**Comment by takagi** [comment]
We usually import modules on top of files whenever possible. Would you move importing  SciPy there? As SciPy is optional for CuPy's test, it should look like this (scipy.special instead of scipy.signal).

https://github.com/cupy/cupy/blob/c8bcae476916985074f5dca4a9f6a5d10458fccd/tests/cupyx_tests/scipy_tests/signal_tests/test_splines.py#L10-L13

It looks good to me other than that. Thanks!

**Comment by chettub** [comment]
@takagi Fixed it!

**Review by takagi — APPROVED** [review]
LGTM!

File: tests/cupyx_tests/scipy_tests/special_tests/test_spherical_bessel.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from cupy import testing
+import cupyx.scipy.special  # NOQA
+
+try:
+    import scipy.special  # NOQA
+except ImportError:
+    pass
+
+
+@testing.gpu
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/special_tests/test_spherical_bessel.py:12]: This decorator was here due to a historical reason and was removed in #7453.
```suggestion
```

**Comment by chettub** [comment]
LGMT!

**Comment by takagi** [comment]
Thanks, @chettub!

**Pull request merged**